Fabric.Authorization.Domain/Services/SecurableItemService.cs

```diff
@@ -46,8 +63,13 @@ public SecurableItemService(IClientStore clientStore)
             item.Id = Guid.NewGuid();
             item.ClientOwner = clientId;
             var client = await _clientStore.Get(clientId);
-            var parentSecurableItem = GetSecurableItemById(client.TopLevelSecurableItem, itemId);
+            var parentSecurableItem = await GetSecurableItem(client.Id, itemId);
             CheckUniqueness(parentSecurableItem, item);
+
+            if (parentSecurableItem.Grain != item.Grain)
+            {
+                throw new BadRequestException<SecurableItem>("The SecurableItem child grain must match the parent SecurableItem's grain.");
```

Comment:
Should the Domain know this is a BadRequest? Should we make this an exception that is domain specific, and not REST API specific?

Reply:
I'm not sure. The exception is a domain-level exception, and this and other services throw this exception along with `AlreadyExistsException` and `NotFoundException`. Did you want to rename `BadRequestException` to something else?

Reply:
We can rename it in a future PR - just don't want HTTP concepts leaking into the domain.